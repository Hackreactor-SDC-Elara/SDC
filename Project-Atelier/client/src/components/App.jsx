import React from 'react';
import Overview from './Overview.jsx';
import RatingsReviews from './ratingsReviews/RatingsReviews.jsx';
import QuestionsAnswers from './QuestionsAnswers.jsx';
import RelatedItems from './RelatedItems.jsx';
import {addOutfitListToCookie, getOutfitListInCookie, deleteOutfitIdInCookie} from '../../../helpers/helpers.js';


export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      product_id: new URLSearchParams(window.location.search).get('id'),
      cart: {},
      outfit: getOutfitListInCookie() === undefined ? [] : getOutfitListInCookie(),
    }
    this.addToCart = this.addToCart.bind(this);
    this.removeFromCart = this.removeFromCart.bind(this);
    this.addToOutfit = this.addToOutfit.bind(this);
    this.removeFromOutfit = this.removeFromOutfit.bind(this);
  }

  addToCart(sku, qty) {
    let currentQuantityInCart = this.state.cart[sku] || 0;
    this.setState({
      cart: {
        ...this.state.cart,
        [sku]: currentQuantityInCart + qty
      }
    });
  }

  // TODO - delete item if none left in cart...currently the cart will hold
  // the sku with a quantity of 0
  removeFromCart(sku, qty) {
    if (!this.state.cart[sku]) return;
    let newQuantity = this.state.cart[sku] - qty;
    newQuantity = newQuantity > 0 ? newQuantity : 0;
    this.setState({
      cart: {
        ...this.state.cart,
        [sku]: newQuantity
      }
    });
  }

  addToOutfit(product_id) {
    // console.log(`adding to outfit: ${product_id}`);
    addOutfitListToCookie(product_id);
    this.setState({outfit: getOutfitListInCookie()});
  }

  removeFromOutfit(product_id) {
    // console.log(`removing from outfit: ${product_id}`);
    this.setState({outfit: deleteOutfitIdInCookie(product_id)})
  }

  render() {
    return (
      <div>
        <Overview
          productId={this.state.product_id}
          addToCart={this.addToCart}
          addToOutfit={this.addToOutfit}
          removeFromOutfit={this.removeFromOutfit}
          inOutfit={this.state.outfit.includes(this.state.product_id)}
        />
        <RelatedItems
          productId={this.state.product_id}
          outfit={this.state.outfit}
          addToOutfit={this.addToOutfit}
          removeFromOutfit={this.removeFromOutfit}
        />
        <QuestionsAnswers productId={this.state.product_id} />
        <RatingsReviews productId={this.state.product_id} />
      </div>
    );
  }
};