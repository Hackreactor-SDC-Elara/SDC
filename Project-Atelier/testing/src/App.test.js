import React from "react";
import render from "react-test-renderer";
import App from "../../client/src/components/App.jsx";


describe("App Component", function () {
  it("should have hello world message", function () {
    console.log(App);
    console.log(render.create(<App />));
    // const component = renderer.create(
    //   <App page="http://www.facebook.com">Facebook</App>,
    // );
    // let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    // let getByText = renderer(<App />);
    // expect(getByText("Hello World")).toMatchInlineSnapshot(`
    //   <h1>
    //     Hello world
    //   </h1>
    // `);
  });
});