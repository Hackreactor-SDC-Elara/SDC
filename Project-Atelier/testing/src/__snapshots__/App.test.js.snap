// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should have hello world message 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <div>
    <h2>
      Review Section
    </h2>
    <h3>
      These are hard coded reviews.
    </h3>
    <div>
      <ul>
        <li>
          Comfortable and practical.
        </li>
        <li>
          They are very dark. But that's good because I'm in very sunny spots
        </li>
      </ul>
    </div>
  </div>
</div>
`;
